cmd/config-manager/main.go

```diff
@@ -165,9 +191,73 @@ func updateContainerdConfig(config map[string]interface{}) map[string]interface{
 	}
 
 	nri["disable"] = false
+
+	cfg.updateContainerdConfig(config)
+
 	return config
 }
 
+func (cfg *nriConfig) check() error {
+	switch {
+	case cfg.registrationTimeout == "" && cfg.requestTimeout == "":
+		return nil
+	case cfg.registrationTimeout != "" && cfg.requestTimeout == "":
+		return fmt.Errorf("NRI plugin registration timeout set without request timeout")
+	case cfg.registrationTimeout == "" && cfg.requestTimeout != "":
+		return fmt.Errorf("NRI plugin request timeout set without registration timeout")
+	}
+
+	register, err := time.ParseDuration(cfg.registrationTimeout)
+	if err != nil {
+		return fmt.Errorf("invalid plugin registration timeout: %w", err)
+	}
+	request, err := time.ParseDuration(cfg.requestTimeout)
+	if err != nil {
+		return fmt.Errorf("invalid plugin request timeout: %w", err)
+	}
+
+	if register <= request {
+		return fmt.Errorf("NRI plugin registration timeout (%s) must be > request timeout (%s)",
+			register, request)
+	}
+
+	return nil
+}
+
+func (cfg *nriConfig) writeCrioConfig(f *os.File) error {
+	const (
+		registrationTimeout = "nri_plugin_registration_timeout"
+		requestTimeout      = "nri_plugin_request_timeout"
+	)
+	if cfg.registrationTimeout != "" {
```

Comment:
Technically it is possible to put even zero right now, since we don't have validation that would block passing 0 value. Zero value could be a problematic and should not be allowed I would assume. Should we perhaps check weather value is non zero before entering the condition ?

Reply:
Yes, we probably should do some further checking. We already check that the registration timeout is > than the request timeout. Maybe you were right in your earlier comment and we should not expect timeouts in time.ParseDuration-accepted formats but instead restrict it much more strictly. For instance allow only seconds to be given and document this, then don't accept units in the setting, only integers and use a minimum and a maximum in the verification schema to limit the timeouts to some reasonable range, for instance 1 sec <= requestTimeout <= 5 sec and 5 sec <= registrationTimeout <= 15 sec...

Would that sound reasonable ? I really would like to get some basic value verification in the schema, so that instead of creating the pod of the deployment failing, creating the deployment would fail before trying to schedule any pods. It is much more user friendly than having to use kubectl get * to dig out what might be the reason of the pods failing.

Reply:
@fmuyassarov I ended up using the values JSON schema for limiting the timeouts and for ensuring that both (or neither) are given. The registration timeout is now limited to `5-30s` and the request timeout is `2-30s`.

That leaves us with the slightly user-unfriendly runtime check of `requestTimeout < registrationTimeout`, but I don't think the JSON schema would be expressive enough to allow expressing such a constraint, so this is the best I can do.